posthog/temporal/data_imports/sources/linkedin_ads/schemas.py

```diff
@@ -0,0 +1,153 @@
+from enum import StrEnum
+from typing import NotRequired, TypedDict
+
+from posthog.warehouse.types import IncrementalFieldType
+
+# Field type mappings for data type conversion
+INTEGER_FIELDS = {
+    "id",
+    "campaign_id",
+    "campaign_group_id",
+    "impressions",
+    "clicks",
+    "externalWebsiteConversions",
+    "landingPageClicks",
+    "totalEngagements",
+    "videoViews",
+    "videoCompletions",
+    "oneClickLeads",
+    "follows",
+}
+FLOAT_FIELDS = {"costInUsd"}
+DATE_FIELDS = {"dateRange.start"}
+
+# Virtual column mappings for analytics resources
+VIRTUAL_COLUMNS = {"campaign_id", "campaign_group_id"}
+
+# This maps the virtual column name to the URN type. LinkedIn Ads API uses URNs to identify resources.
+# URNs are like "urn:li:sponsoredCampaign:185129613"
+VIRTUAL_COLUMN_URN_MAPPING = {"campaign_id": "Campaign", "campaign_group_id": "CampaignGroup"}
+RESOURCE_VIRTUAL_COLUMNS = {"campaign_stats": "campaign_id", "campaign_group_stats": "campaign_group_id"}
+
+
+class LinkedinAdsResource(StrEnum):
+    Accounts = "accounts"
+    Campaigns = "campaigns"
+    CampaignGroups = "campaign_groups"
+    CampaignStats = "campaign_stats"
+    CampaignGroupStats = "campaign_group_stats"
+
+
+class LinkedinAdsPivot(StrEnum):
+    ACCOUNT = "ACCOUNT"
+    CAMPAIGN = "CAMPAIGN"
+    CAMPAIGN_GROUP = "CAMPAIGN_GROUP"
+    CREATIVE = "CREATIVE"
+
+
+# LinkedIn API endpoint mappings
+LINKEDIN_ADS_ENDPOINTS = {
+    LinkedinAdsResource.Accounts: "adAccounts",
+    LinkedinAdsResource.Campaigns: "adCampaigns",
+    LinkedinAdsResource.CampaignGroups: "adCampaignGroups",
+    LinkedinAdsResource.CampaignStats: "adAnalytics",
+    LinkedinAdsResource.CampaignGroupStats: "adAnalytics",
+}
+
+# Pivot mappings for analytics resources
+LINKEDIN_ADS_PIVOTS = {
+    LinkedinAdsResource.CampaignStats: LinkedinAdsPivot.CAMPAIGN,
+    LinkedinAdsResource.CampaignGroupStats: LinkedinAdsPivot.CAMPAIGN_GROUP,
+}
+
+
+class ResourceSchema(TypedDict):
+    resource_name: str
+    field_names: list[str]
+    primary_key: list[str]
+    filter_field_names: NotRequired[list[tuple[str, IncrementalFieldType]]]
+
+
+# LinkedIn Ads resource schemas
+RESOURCE_SCHEMAS: dict[LinkedinAdsResource, ResourceSchema] = {
+    LinkedinAdsResource.Accounts: {
+        "resource_name": "accounts",
+        "field_names": ["id", "name", "status", "type", "currency", "version"],
+        "primary_key": ["id"],
+    },
+    LinkedinAdsResource.Campaigns: {
```

Comment:
There are a bunch of fields missing from here, such as the `changeAuditStamps` - this could be used as the partition key - https://learn.microsoft.com/en-us/linkedin/marketing/integrations/ads/account-structure/create-and-manage-campaigns?view=li-lms-2025-08&tabs=http#search-for-campaigns

<img width="393" height="179" alt="image" src="https://github.com/user-attachments/assets/05ba6dfd-bf71-48e8-a8be-5438856b417f" />

Reply:
Yeah, I'd just pull these out into a `created_at` and `updated_at` field